src/schema/slack/functions/_scripts/src/templates/test_template.ts

```diff
@@ -0,0 +1,137 @@
+import { manifestFunctionFieldsToTypeScript } from "./template_function.ts";
+import SchemaTypes from "../../../../../schema_types.ts";
+
+export const manifestFunctionToTypeScript = (
+  functionRecord: FunctionRecord,
+) => {
+  return `{${manifestFunctionFieldsToTypeScript(functionRecord)}}`;
+};
+
+const renderFunctionManifestTest = (
+  functionRecord: FunctionRecord,
+) => {
+  const functionName = getFunctionName(functionRecord.callback_id);
+  const typescript: string[] = [];
+  typescript.push(
+    `assertEquals(${functionName}.definition.callback_id, "${
+      getSlackCallbackId(functionRecord)
+    }");`,
+  );
+  typescript.push(
+    `const expected: ManifestFunctionSchema = ${
+      manifestFunctionToTypeScript(
+        functionRecord,
+      )
+    };`,
+  );
+  typescript.push(`const actual = ${functionName}.export();`);
+  typescript.push("");
+  typescript.push(`assertEquals(actual, expected);`);
+  return `() => {${typescript.join("\n")}}`;
+};
+
+const workflowToTypeScript = (functionName: string) => {
+  const typescript: string[] = [];
+  typescript.push(`callback_id: "test_${functionName}_built_in"`);
+  typescript.push(`title: "Test ${functionName}"`);
+  typescript.push(
+    `description: "This is a generated test to test ${functionName}"`,
+  );
+  return `{${typescript.join(", \n")}}`;
+};
+
+const getInputValue = (name: string, type: string) => {
+  switch (type) {
+    case SchemaTypes.integer:
+      return `${name}: 1`;
+    case SchemaTypes.boolean:
+      return `${name}: true`;
+    case SchemaTypes.number:
+      return `${name}: 1.5`;
+    case SchemaTypes.array:
+      return `${name}: []`;
+    case SchemaTypes.object:
+      return `${name}: {}`; // TODO make this recursive
+    default:
+      return `${name}: "test"`;
+  }
+};
+
+const requiredParametersToTypeScript = (
+  parameters: FunctionParameter[],
+) => {
+  const typescript: string[] = [];
+  parameters.forEach((parameter: FunctionParameter) => {
+    if (parameter.is_required) {
+      typescript.push(getInputValue(parameter.name, parameter.type));
+    }
+  });
+  return `{${typescript.join(",\n")}}`;
+};
+
+const renderWorkflowStepTest = (dfi: FunctionRecord) => {
+  const functionName = getFunctionName(dfi.callback_id);
+  const inputParameters = requiredParametersToTypeScript(
+    dfi.input_parameters,
+  );
+  const typescript: string[] = [];
+  typescript.push(
+    `const testWorkflow = DefineWorkflow(${
+      workflowToTypeScript(functionName)
+    });`,
+  );
+  typescript.push(
+    `testWorkflow.addStep(${functionName}, ${inputParameters});`,
+  );
+  typescript.push(`const actual = testWorkflow.steps[0].export();`);
+  typescript.push("");
+  typescript.push(
+    `assertEquals(actual.function_id, "${getSlackCallbackId(dfi)}");`,
+  );
+  typescript.push(`assertEquals(actual.inputs, ${inputParameters});`);
+  return `() => {${typescript.join("\n")}}`;
+};
+
+export function SlackTestFunctionTemplate(
```

Comment:
My opinion is that the most valuable things we can test for built-ins is
- the optionality for their step inputs (making sure that all required fields must be set, making sure non-required field can be set but are dont have to be)
- that we can access all outputs regardless of if they are required or not

There are lots of combinations we could generate tests for in this case, but here's an idea of what this could look like for `ArchiveChannel`
```ts
/** testing inputs */
const step = testWorkflow.addStep(ArchiveChannel, { channel_id: "test" });
// @ts-expect-error not providing `channel_id` leads to a type error
testWorkflow.addStep(ArchiveChannel, {});

/** testing outputs */
assert<CannotBeUndefined<typeof step.outputs.channel_id>>(true);
```

Reply:
I've implemented a new generated test to cover outputs 💯